### PR TITLE
#5383 close subscribe popover by clicking anywhere

### LINKED ISSUE #5383: Close the "Subscribe+" popover  by clicking anywhere on the screen
Hi, this is a first-timers-only issue. This means we've worked to make it more legible to folks who either haven't contributed to our codebase before, or even folks who haven't contributed to open source before.

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, consider leaving this one for someone new, and looking through our general help wanted issues. Thanks!

🤔 **What you will need to know.**
Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

📋 **Step by Step**
🙋 Claim this issue: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

📝 Update the file [app/views/tag/_subscribe_button.html.erb] (https://github.com/publiclab/plots2/tree/master/app/views/tag/_subscribe_button.html.erb) in the plots2 repository. Currently, the subscribe popover closes only by clicking it back on it.

Attach a click event to the document body excluding the "#mypopcontent" which can be found by getElementById(), which closes the popover. 

💾 Commit your changes

🔀 Start a Pull Request. There are two ways how you can start a pull request:

If you are familiar with the terminal or would like to learn it, here is a great tutorial on how to send a pull request using the terminal.

You can also edit files directly in your browser and open a pull request from there.

🏁 **Done** Ask in comments for a review :)
Please keep us updated
💬⏰ - We encourage contributors to be respectful to the community and provide an update within a week of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance!

If this happens to you, don't sweat it! Grab another open issue.

**Is someone else already working on this?**
🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it.

👥- If someone seems stuck, offer them some help! Otherwise, take a look at some other issues you can help with. Thanks!

🤔❓ **Questions?**
Leave a comment below!

### PR DESCRIPTION
**Pull request by ckairen**
Fixes #5383 (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below
popover_fix_min

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

File: app/views/tag/_subscribe_button.html.erb [hunk 1]
```diff
@@ -24,10 +24,26 @@
...
+  $('#popover_button').popover({
+  	content: $('#mypopcontent').clone().removeClass('hidden'),
+  	placement: 'bottom',
+  	html: true,
+
+  }).click(function(e) {
+    e.stopPropagation();
+  });
+
+  $(document).click(function(e){
```
Comment (siaw23) [inline app/views/tag/_subscribe_button.html.erb]: I think this is the only relevant code. Why do we need lines 33-45? Can you please check this and remove what we don't need?

Comment (ckairen) [reply app/views/tag/_subscribe_button.html.erb]: line 46,47 hides the popover whenever we click anywhere on the document. However, this is not the desired functionality. We want to exclude the popover window itself and also the subscribe button from this. (we don't want to close the popover when we 1. click on subscribe the first time which should open the popover, 2. when we click on the popover window to select things) Therefore the e.stopPropagation is needed to exclude those two components from executing the "hide" when clicked on it.

**Comment by gauravano** [comment]
Hi @ckairen, I tried your PR locally and the popover is closing when I select a checkbox or click inside popover

inconsistent_popover

**Comment by gauravano** [comment]
Also, we upgraded to Bootstrap4 recently so rebase your branch with master and then see the result. Thanks!

**Comment by SidharthBansal** [comment]
Hi, just checking if you've gotten stuck on this at all, or if I could help in any way? Thanks!

**Comment by SidharthBansal** [comment]
As the person is inactive for more than a month, I am closing the PR. In case you want to push changes please feel free to open a new PR OR reopen this PR and add additional changes to it.
Thanks for contributing on Public Lab

**Pull request closed without merging**